packages/calcite-components/src/utils/dom.ts

```diff
@@ -682,11 +682,25 @@ function nextFrame(): Promise<void> {
 }
 
 /**
- * This util helps determine when a string value is a pixel value.
- *
- * @param value The value to check.
- * @returns {boolean} Whether the value is a pixel value.
- */
-export function isPixelValue(value: string): boolean {
-  return value.endsWith("px");
+ * This helper converts a CSS style value (e.g., "px", "vw", "vh") into its pixel equivalent.
+ *
+ * - If the value ends with "px", it parses and returns the numeric value.
+ * - If the value ends with "vw", it calculates the pixel value based on the viewport width.
+ * - If the value ends with "vh", it calculates the pixel value based on the viewport height.
+ * - For unsupported units or invalid values, it returns 0.
+ *
+ * @param {string} value - The CSS style value to convert (e.g., "10px", "50vw", "30vh").
+ * @returns {number} The pixel equivalent of the provided value.
+ */
+export function getStylePixelValue(value: string): number {
```

Comment:
Future support could include `svh`, `dvh`, etc. as needed for improved device support if devs opt for that.